Examples/SwiftOpenAIExample/SwiftOpenAIExample/AIProxyIntroView.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  AIProxyIntroView.swift
+//  SwiftOpenAIExample
+//
+//  Created by Lou Zell on 3/27/24.
+//
+
+import SwiftUI
+import SwiftOpenAI
+
+struct AIProxyIntroView: View {
+
+   @State private var partialKey = ""
+   @State private var deviceCheckBypass = ""
+
+   var body: some View {
+      NavigationStack {
+         VStack {
+            Spacer()
+            VStack(spacing: 24) {
+               TextField("Enter partial key", text: $partialKey)
+               TextField("Enter DeviceCheck bypass", text: $deviceCheckBypass)
+            }
+            .padding()
+            .textFieldStyle(.roundedBorder)
+
+            NavigationLink(destination: OptionsListView(openAIService: aiproxyService)) {
+               Text("Continue")
+                  .padding()
+                  .padding(.horizontal, 48)
+                  .foregroundColor(.white)
+                  .background(
+                     Capsule()
+                        .foregroundColor(partialKey.isEmpty ? .gray.opacity(0.2) : Color(red: 64/255, green: 195/255, blue: 125/255)))
+            }
+            .disabled(partialKey.isEmpty)
+            Spacer()
+            Group {
+               Text("You can now use SwiftOpenAI for development and production! AI Proxy keeps your OpenAI API key secure. To configure AI Proxy for your project, or to learn more about how it works, please see the docs at ") + Text("[this link](https://www.aiproxy.pro/docs).")
+            }
+            .font(.caption)
+         }
+         .padding()
+         .navigationTitle("AIProxy Configuration")
+      }
+   }
+
+   private var aiproxyService: some OpenAIService {
+      // Attention AI Proxy customers!
```

Comment:
Yup, for sure! I will work up a PR for the README today to satisfy the second bullet here: https://github.com/jamesrochabrun/SwiftOpenAI/pull/20#pullrequestreview-1964062461

Reply:
What if we move this conditional inside the factory? like., and make this the only way to create an AIProxy service? 

```swift
public static func service(
   aiproxyPartialKey: String,
   aiproxyDeviceCheckBypass: String?) -> OpenAIService 
{
     if let aiproxyDeviceCheckBypass {
            #if DEBUG && targetEnvironment(simulator)
              /// return service 
            #else
            // return service 
          #endif
       } else  {
       // return service
      }
}
```

Reply:
The purpose of these compiler conditionals is, I admit, confusing.

If the bypass is compiled into the binary, it's completely fine to pass around (so, for example, if someone called the factory with both the `aiproxyPartialKey` and `aiproxyDeviceCheckBypass`, it would be fine for us to pass those to the AIProxyService).

What I am trying to prevent is the device check bypass entering the binary for a build that will be distributed. Because once in the binary, it's trivial to dump using `strings` or many other tools. Here is an example of a friend's app (I told him about this):

<img width="686" alt="AspectShot-2024-03-26T09 40 10" src="https://github.com/jamesrochabrun/SwiftOpenAI/assets/35940/5330298b-8ae9-45ea-aa75-58d460ae0006">

So, the challenge is to keep the token out of the binary itself, and for that I need to anticipate how developers will use this library and sample app. I _think_ a common use case will be to fire up the sample app and then copy/paste the call site that the developer needs for integrating it into their own app. This is why I wrote the note and put the conditionals where I did, hoping that a developer copying/pasting the call to the factory into their own app would also include the compiler conditionals.

Does that make sense? This is admittedly a concern that we don't often encounter. I'm still working out the best way to address it. The goal, though, is to keep that token out of the compiled binary for a distribution build. If it's already in the binary, then it's fine to pass around in memory.

Reply:
Thanks for adding those comments, I have to admit this goes beyond my area of expertise but I think makes sense, lets keep the README clear about this. Thanks!